sdk/tables/data-tables/src/TablesBatch.ts

```diff
@@ -0,0 +1,321 @@
+    createPipeline() {
+      const policyFactoryLength = 3;
+      const factories: RequestPolicyFactory[] = new Array(policyFactoryLength);
+
+      factories[0] = deserializationPolicy();
+      factories[1] = new BatchHeaderFilterPolicyFactory();
+
+      factories[policyFactoryLength - 1] = new BatchRequestAssemblePolicyFactory(this); // Use batch assemble policy to assemble request and intercept request from going to wire
+      return factories;
+    },
+    appendSubRequestToBody(request: WebResource) {
+      // Start to assemble sub request
+      this.body += [
+        subRequestPrefix, // sub request constant prefix
+        "", // empty line after sub request's content ID
+        `${request.method.toString()} ${getURLPathAndQuery(
+          request.url
+        )} ${HTTP_VERSION_1_1}${HTTP_LINE_ENDING}` // sub request start line with method,
+      ].join(HTTP_LINE_ENDING);
+
+      for (const header of request.headers.headersArray()) {
+        this.body += `${header.name}: ${header.value}${HTTP_LINE_ENDING}`;
+      }
+
+      this.body += `${HTTP_LINE_ENDING}${request.body}${HTTP_LINE_ENDING}`; // sub request's headers need end with an empty line
+      this.operationCount++;
+    },
+    getHttpRequestBody(): string {
+      return `${this.body}${HTTP_LINE_ENDING}${chaangesetEnding}${HTTP_LINE_ENDING}${batchEnding}${HTTP_LINE_ENDING}`;
+    },
+    getMultipartContentType(): string {
+      return multipartContentType;
+    },
+    getBatchBoundary(): string {
+      return batchBoundary;
+    }
+  };
+}
+
+/**
+ * Get URL path and query from an URL string.
+ *
+ * @export
+ * @param {string} url Source URL string
+ * @returns {(string | undefined)}
+ */
+export function getURLPathAndQuery(url: string): string | undefined {
+  const urlParsed = URLBuilder.parse(url);
+  const pathString = urlParsed.getPath();
+  if (!pathString) {
+    throw new RangeError("Invalid url without valid path.");
+  }
+
+  let queryString = urlParsed.getQuery() || "";
+  queryString = queryString.trim();
+  if (queryString !== "") {
+    queryString = queryString.startsWith("?") ? queryString : `?${queryString}`; // Ensure query string start with '?'
+  }
+
+  return `${urlParsed.getScheme()}://${urlParsed.getHost()}${urlParsed.getPath()}`;
+}
+
+class BatchHeaderFilterPolicy extends BaseRequestPolicy {
+  // eslint-disable-next-line @typescript-eslint/no-useless-constructor
+  constructor(nextPolicy: RequestPolicy, options: RequestPolicyOptions) {
+    super(nextPolicy, options);
+  }
+  public async sendRequest(request: WebResourceLike): Promise<HttpOperationResponse> {
+    let xMsHeaderName = "";
+
+    for (const header of request.headers.headersArray()) {
+      if (header.name.toLowerCase() === HeaderConstants.X_MS_VERSION) {
+        xMsHeaderName = header.name;
+      }
+    }
+
+    if (xMsHeaderName !== "") {
+      request.headers.remove(xMsHeaderName); // The subrequests should not have the x-ms-version header.
+    }
```

Comment:
You can use `contains` to check, but you really don't have to. Operations on `HttpHeaders` are case insensitive anyway.

```suggestion
    request.headers.remove(HeaderConstants.X_MS_VERSION); // The subrequests should not have the x-ms-version header.
```